api/models.py

```diff
@@ -117,6 +117,25 @@ def to_json(self):
             "duration": self.duration,
         }
 
+    def to_json(self):
+        # Extract the list of answers from the new data structure
+        incorrect_answers = []
+        for item in self.incorrect_answer_list:
+            if isinstance(item, dict) and 'answer' in item:
+                incorrect_answers.append(item['answer'])
+            else:
+                # In case of any anomalies, include the item as is
+                incorrect_answers.append(item)
```

Comment:
Very nice; we need to add logging in the `else` body; I'll add a GitHub issue from this comment!